insights/parsers/initscript.py

```diff
@@ -0,0 +1,84 @@
+"""
+InitScript - files ``/etc/rc.d/init.d/*``
+=========================================
+
+InitScript is a parser for the initscripts in ``/etc/rc.d/init.d``.
+
+Because this parser read multiple files, the initscripts are stored as a list
+within the parser and need to be iterated through in order to find specific
+initscripts.
+
+Examples:
+
+    >>> for initscript in shared[InitScript]: # Parser contains list of all initscripts
+    ...     print "Name:", initscript.file_name
+    ...
+    Name: rhnsd
+
+"""
+
+from . import SkipException
+from .. import Parser, parser
+import re
+
+CHKCONFIG_REGEX = re.compile(r"^#\s+chkconfig:\s+.*$")
+LSB_REGEX = re.compile(r"^#\s+Provides:\s+.*$")
+
+SHEBANG_REGEX = re.compile(r"^#!\s*/.*$")
+STARTSTOPSTATUS_REGEX = re.compile(r"\b(start|stop|status)\b")
+
+
+class EmptyFileException(SkipException):
+    pass
+
+
+class NotInitscriptException(SkipException):
+    pass
+
+
+@parser("initscript")
+class InitScript(Parser):
+    """
+    Parse initscript files. Each item is a dictionary with following fields:
```

Comment:
@rmetrich would you mind to include your description of the algorithm you are implementing here, I think it would be helpful for users of your parser in understanding how it works, and the outputs.  You could start with your PR comments and add if you feel like anything needs further explanation.

Reply:
@bfahr Done